File: 1791.找出星型图的中心节点.py
#
# @lc app=leetcode.cn id=1791 lang=python3
#
# [1791] 找出星型图的中心节点
#

# @lc code=start
class Solution:
    def findCenter(self, edges: List[List[int]]) -> int:
        m = len(edges) + 1
        temp = [0 for _ in range(m)]
        for edge in edges:
            temp[edge[0] - 1] += 1
            temp[edge[1] - 1] += 1
        
        res = 0
        for i in range(m):
            if temp[i] == m - 1:
                res = i + 1
                break
        return res
# @lc code=end

File: 1791.找出星型图的中心节点.go
/*
 * @lc app=leetcode.cn id=1791 lang=golang
 *
 * [1791] 找出星型图的中心节点
 */

// @lc code=start
func findCenter(edges [][]int) (res int) {
	m := len(edges) + 1
	tempSlice := make([]int, m)
	for _, edge := range edges {
		tempSlice[edge[0] - 1]++
		tempSlice[edge[1] - 1]++
	}

	for idx, num := range tempSlice {
		if num == m - 1 {
			res = idx + 1
			break
		}
	}
	return
}
// @lc code=end

